llvm/include/llvm/Transforms/Utils/SCCPSolver.h

```diff
@@ -151,6 +151,10 @@ class SCCPSolver {
   /// works with both scalars and structs.
   void markOverdefined(Value *V);
 
+  /// trackValueOfArgument - Mark the specified argument overdefined unless it
+  /// have range attribute.  This works with both scalars and structs.
```

Comment:
>  This works with both scalars and structs.

I don't understand why it works with structs. Doesn't it only work with integer scalars?

Reply:
as it calls the markOverdefined if it is not a integer with a range attribute and that function have that text I selected to have it for this function also.